core/src/main/java/hudson/DNSMultiCast.java

```diff
@@ -1,205 +1,21 @@
 package hudson;
 
 import jenkins.util.SystemProperties;
-import jenkins.model.Jenkins;
-import jenkins.model.Jenkins.MasterComputer;
 
-import javax.jmdns.ServiceInfo;
-import javax.jmdns.impl.JmDNSImpl;
 import java.io.Closeable;
 import java.io.IOException;
-import java.lang.reflect.Field;
-import java.lang.reflect.InvocationTargetException;
-import java.lang.reflect.Method;
-import java.net.InetAddress;
-import java.net.URL;
-import java.util.HashMap;
-import java.util.Map;
-import java.util.concurrent.Callable;
-import java.util.concurrent.ExecutorService;
-import java.util.logging.Level;
-import java.util.logging.Logger;
 
 /**
  * Registers a DNS multi-cast service-discovery support.
  *
  * @author Kohsuke Kawaguchi
+ * @deprecated
  */
+@Deprecated
 public class DNSMultiCast implements Closeable {
-    private JenkinsJmDNS jmdns;
-
-    public DNSMultiCast(final Jenkins jenkins) {
```

Comment:
You are deleting the original constructor so is there some reason you are even keeping this class around?

Reply:
The Jenkins Test Harness references the `disabled` property. There could be other places. For now at least I left the class to not break the JTH, but it doesn't do anything because it is always disabled. I thought about tying this to a change in the JTH but wasn't sure we wanted to connect the versions like that. WDYT?

Reply:
> The Jenkins Test Harness references the `disabled` property.

OK, this was the information I was missing.

> wasn't sure we wanted to connect the versions like that

Indeed we do not.

Reply:
also https://github.com/jenkinsci/jenkinsfile-runner/blob/c267cbe255ed81668187e50378a0ea12407cb5aa/setup/src/main/java/io/jenkins/jenkinsfile/runner/JenkinsEmbedder.java#L682